### ISSUE TITLE
Add ability to customize store UI (theming support)

### ISSUE THREAD

**Issue by MrNaif2018** [bitcartcc/bitcart-store#319]
Also tracked in bitcartcc/bitcart#166

We need to add ability to customize theme colors of the store (for custom brands). Probably the best solution is to use css variables.

I have implemented basic theming support in 3730413c3cae25244b5ab28656401c646b1a9ad6 and 1d023187d32fa08bf4043badc2137e6b4a18979b

There are few things left to make it fully customizeable:

- [ ] Make all custom bulma libraries (bulma-steps, bulma-switch, etc) use same theme settings
Currently for example switch color is still green even when success color of the theme was changed to another color, see:
https://github.com/Wikiki/bulma-switch/blob/master/src/sass/index.sass#L1
First of all to make it all work, we need to remove stylus and use sass everywhere, see #316
After that we need to ensure that sass customizations get applied to component scoped styling.
If not all components styling can be moved to `main.sass` file
https://github.com/bitcartcc/bitcart-store/blob/1d023187d32fa08bf4043badc2137e6b4a18979b/components/Switch.vue#L25
Also probably we should import not from css, but from source sass files to be able to customize some variables
If it still doesn't work, we might need to expose more colors to customization or change switch-background variable to use some color like $primary instead of $grey-light
- [ ] Add more colors and variables for customization if needed:
https://github.com/bitcartcc/bitcart-store/blob/1d023187d32fa08bf4043badc2137e6b4a18979b/bulma-css-vars.config.js#L4-L17
Also maybe there could be a better way to not hardcode hsl values of colors
- [ ] Add ability to load custom theme css. Sample file:
```
:root {
  --brand-color: #162d50;
  --primary: var(--brand-color) !important;
  --success: var(--brand-color) !important;
  --link: var(--brand-color) !important;
}
```
The css should be loaded dynamically via `head` function inside default layout.
Css file url itself is a variable got from store policies.
Merchants API needs to be customized to add new store theming settings
- [ ] And if all of the above is implemented, we could try to allow using [bulmaswatch](https://jenil.github.io/bulmaswatch/) for easily selecting themes. This means that some existing styles might need to be customized to allow this

Note that all themes being applied should happen at the runtime, so at sass compilation step we should provide all needed css variables for modification later

**Comment by beatrizsmerino** [comment]
thanks !! :)
I'm reading the requirements and making a fork of the project

**Comment by beatrizsmerino** [comment]
![image](https://user-images.githubusercontent.com/14045148/136713967-99445517-6a66-4d55-8274-5b13788d6295.png)
I'm having problems for run the project `npm run dev`

**Comment by MrNaif2018** [comment]
Yeah sorry for that, it needs to be fixed. There is an open issue for that, but I was waiting if someone would take it up (:
https://github.com/bitcartcc/bitcart-store/issues/318
See https://github.com/bitcartcc/bitcart-store/issues/316#issuecomment-932450528 this guide on how to set up merchants api
Also we use `yarn`, it is recommended to not mix package managers' lock files

**Comment by MrNaif2018** [comment]
Let me know if you have any issues.
I have documented how to run admin panel here:
https://github.com/bitcartcc/bitcart-admin/blob/master/CONTRIBUTING.md#merchants-api
For store, replace `BITCART_ADMIN_API_URL` with `BITCART_STORE_API_URL`

**Comment by MrNaif2018** [comment]
Hi @beatrizsmerino! Have you been able to look at this? Do you need some help?

**Comment by beatrizsmerino** [comment]
I haven't been able to get on it yet, I've been busy, but I will do it this week.

**Comment by beatrizsmerino** [comment]
In the end I don't have time to do it, sorry.

**Comment by MrNaif2018** [comment]
Sure. It will get implemented by itself gradully
